lib/checkstl.cpp

```diff
@@ -866,6 +879,8 @@ void CheckStl::mismatchingContainerIterator()
             ValueFlow::Value val = getLifetimeIteratorValue(iterTok);
             if (!val.tokvalue)
                 continue;
+            if (!val.isKnown() && Token::simpleMatch(val.tokvalue->astParent(), ":"))
```

Comment:
The lifetime values are never known. Also, this will lead to FNs as well. I think it would be better to check that the set of lifetimes are the same when there are multiple lifetimes.

Reply:
Isn't the problem that there is only one value? Otherwise, `getLifetimeIteratorValue()` would return an empty value.

Reply:
Ah yea, thats right, but it should be multiple values. I wonder why we are not getting both lifetime.

I assumed the error is that one variable has the lifetime `v1` and the other has the lifetime `v2`. It should have both lifetimes.